### PR TITLE
Add a test for the select/deselect all functionality

### PR DESCRIPTION
**Pull request by Abestanis**
This adds test for the fix in #138.

To prove that these tests actually test against the problem fixed by #138, look at the test run [here](https://github.com/nt4f04uNd/sweyer/actions/runs/9930024597/job/27428273148?pr=142) against master which doesn't have the fix from #138.

### CODE REVIEW

File: test/routes/selection_route_test.dart [hunk 1]
```diff
@@ -262,4 +263,73 @@ void main() {
...
+    final playlist2 = playlistWith(id: 2, songIds: [song2.id]);
+
+    for (var tabAndContentType in [
+      const Tuple2('tracks', ContentType.song),
+      const Tuple2('album', ContentType.album),
+      const Tuple2('playlists', ContentType.playlist),
+      const Tuple2('artists', ContentType.artist),
+    ]) {
+      var tabName = tabAndContentType.item1;
+      var contentType = tabAndContentType.item2;
```
Comment (nt4f04uNd) [inline test/routes/selection_route_test.dart:288]: nit: this could be final

File: test/routes/selection_route_test.dart [hunk 2]
```diff
@@ -262,4 +263,73 @@ void main() {
...
+      testWidgets('allows selecting all and deselecting all on the $tabName tab', (WidgetTester tester) async {
+        await setUpAppTest(() {
+          FakeSweyerPluginPlatform.instance.songs = [song0, song1, song2];
+          FakeSweyerPluginPlatform.instance.albums = [album0, album1, album2];
+          FakeSweyerPluginPlatform.instance.playlists = [playlist0, playlist1, playlist2];
+          FakeSweyerPluginPlatform.instance.artists = [artist0, artist1, artist2];
+        });
+        await tester.runAppTest(() async {
+          // Select tab
+          if (contentType != ContentType.song) {
```
Comment (nt4f04uNd) [inline test/routes/selection_route_test.dart:298]: Initial tab could change, so I would not depend on this condition here

File: test/routes/selection_route_test.dart [hunk 3]
```diff
@@ -262,4 +263,73 @@ void main() {
...
+            await tester.tap(find.byIcon(contentType.icon).first);
+            await tester.pumpAndSettle();
+          }
+
+          // Select first content
+          await tester.longPress(find.byType(ContentTile).first);
+          await tester.pumpAndSettle();
+
+          var selectionController = ContentControl.instance.selectionNotifier.value!;
```
Comment (nt4f04uNd) [inline test/routes/selection_route_test.dart:307]: nit: could be final

File: test/routes/selection_route_test.dart [hunk 4]
```diff
@@ -262,4 +263,73 @@ void main() {
...
+          expect(selectionController.inSelection, true);
+          expect(selectionController.data.length, 1);
+          expect(find.descendant(of: find.byType(SelectionCounter), matching: find.text('1')).hitTestable(),
+              findsOneWidget);
+
+          // Select all
+          await tester.tap(find.byType(SelectAllSelectionAction).last);
+          await tester.pumpAndSettle();
+
+          int numElements = ContentControl.instance.getContent(contentType).length;
```
Comment (nt4f04uNd) [inline test/routes/selection_route_test.dart:317]: nit: could be final

**Review by nt4f04uNd — APPROVED** [review]

**Comment by nt4f04uNd** [comment]
To be clear - this PR was closed automatically

**Comment by nt4f04uNd** [comment]
To be clear - this PR was closed automatically

**Comment by nt4f04uNd** [comment]
Also, I cannot reopen it

**Comment by Abestanis** [comment]
> Also, I cannot reopen it

I can't reopen it either, I'll open a new one with your feedback incorporated.

**Pull request closed without merging**